pkg/scheduler/internal/cache/cache.go

```diff
@@ -708,6 +710,24 @@ func (cache *cacheImpl) addNodeImageStates(node *v1.Node, nodeInfo *framework.No
 	nodeInfo.ImageStates = newSum
 }
 
+// updateNodeImageStates update node record from image entries having the node
+// in imageStates cache when receive node add/update/delete events.
+func (cache *cacheImpl) updateNodeImageStates(newNode *v1.Node) {
+	for existNode := cache.headNode; existNode != nil; existNode = existNode.next {
+		node := existNode.info.Node()
+		if node == nil || node.Name == newNode.Name {
+			continue
+		}
+		newSum := make(map[string]*framework.ImageStateSummary)
+		for image := range cache.imageStates {
+			if cache.imageStates[image].nodes.Has(node.Name) {
+				newSum[image] = cache.createImageStateSummary(cache.imageStates[image])
+			}
+		}
+		existNode.info.ImageStates = newSum
+	}
```

Comment:
I don't quite understand the logic here, you are trying to update the cache with the value get from the cache, shouldn't this do by `addNodeImageStates` or `removeNodeImageStates` instead? Can you explain a little bit how this change fix the issue?

Reply:
`addNodeImageStates` and `removeNodeImageStates ` just update the current node's imageState, but don't update    other node imageState with same image in current node.

Reply:
-1

Every node should be treated independently for two reasons:
- performance: iterating over every node is expensive
- there is no guarantee that two images with the same name will have the same size in two different nodes

I say we close this PR and corresponding issue.

Reply:
> -1
> 
> Every node should be treated independently for two reasons:
> 
> * performance: iterating over every node is expensive
> * there is no guarantee that two images with the same name will have the same size in two different nodes
> 
> I say we close this PR and corresponding issue.

use [Parallelizer](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/parallelize/parallelism.go#L36) to improve performance. Besides, ImageLocality plugin does nothing to do with image size. as for the number of image in node status can use [nodeStatusMaxImages](https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1/) to limit

Reply:
> there is no guarantee that two images with the same name will have the same size in two different nodes

Actually, scratch this. We are already making the assumption that all images have the same size in every node.

However, there is a more performant implementation that doesn't require going over all nodes:
get rid of `imageState` struct in the `cache` package and replace it with `framework.ImageStateSummary` (remove `Summary` from the name). Make sure all nodes point to the same `*framework.ImageState`. That way, you can increase the counter in one node, and it will affect all nodes.

Then, when grabbing a snapshot of the nodes, make sure to copy the values of `map[string]*framework.ImageState` so that we are not sharing the same pointers as the cache. Profit :)

Reply:
> > there is no guarantee that two images with the same name will have the same size in two different nodes
> 
> Actually, scratch this. We are already making the assumption that all images have the same size in every node.
> 
> However, there is a more performant implementation that doesn't require going over all nodes: get rid of `imageState` struct in the `cache` package and replace it with `framework.ImageStateSummary` (remove `Summary` from the name). Make sure all nodes point to the same `*framework.ImageState`. That way, you can increase the counter in one node, and it will affect all nodes.
> 
> Then, when grabbing a snapshot of the nodes, make sure to copy the values of `map[string]*framework.ImageState` so that we are not sharing the same pointers as the cache. Profit :)

thanks for your suggestion, i overridden implementation